### PR TITLE
Add shadow demo for BLE

### PR DESCRIPTION
**Pull request by ravibhagavandas**
Add shadow demo for BLE

Description
-----------
Add demo using new shadow library and BLE transport interface.
Update documentation for shadow and MQTT demos
Change config names

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: demos/ble/mqtt_ble/mqtt_demo_ble_transport.c [hunk 1]
```diff
@@ -31,12 +31,26 @@
...
 #include "core_mqtt.h"
 #include "iot_ble_mqtt_transport.h"
 #include "iot_ble_data_transfer.h"
 
+/*
+ * Demo for showing how to use  the MQTT API with BLE transport interface implementation
+ * to send MQTT packets to cloud assisted by a companion mobile device.
+ *
+ * The Example establishes a BLE data transfer channel over BLE GATT connection with a companion mobile
+ * device and  uses MQTT APIs to create and send  packets over the channel to a broker.
```
Comment (aggarw13) [inline demos/ble/mqtt_ble/mqtt_demo_ble_transport.c:45]:
```suggestion
 * device and uses MQTT APIs to create and send packets over the channel to a broker.
```

File: demos/ble/mqtt_ble/mqtt_demo_ble_transport.c [hunk 2]
```diff
@@ -31,12 +31,26 @@
...
 #include "iot_ble_config.h"
-#include "mqtt_demo_ble_transport_config.h"
+#include "mqtt_ble_demo_config.h"
 
 #include "core_mqtt.h"
 #include "iot_ble_mqtt_transport.h"
 #include "iot_ble_data_transfer.h"
 
+/*
+ * Demo for showing how to use  the MQTT API with BLE transport interface implementation
```
Comment (aggarw13) [inline demos/ble/mqtt_ble/mqtt_demo_ble_transport.c:41]:
```suggestion
 * Demo for showing the use of MQTT API with transport interface implementation based on the BLE connectivity layer
```

File: demos/ble/mqtt_ble/mqtt_demo_ble_transport.c [hunk 3]
```diff
@@ -31,12 +31,26 @@
...
+ * to send MQTT packets to cloud assisted by a companion mobile device.
+ *
+ * The Example establishes a BLE data transfer channel over BLE GATT connection with a companion mobile
+ * device and  uses MQTT APIs to create and send  packets over the channel to a broker.
+ * This example is single threaded and uses statically allocated
+ * memory. It uses QoS0 by default for sending to and receiving messages from the broker.
+ *
+ * A smartphone application using the FreeRTOS companion mobile SDK, in BLE central role should scan and
+ * connect to this device successfully, before starting the demo.
+ * Broker endpoint and the thing name should be configured in aws_clientcredential.h.
```
Comment (aggarw13) [inline demos/ble/mqtt_ble/mqtt_demo_ble_transport.c:51]:
```suggestion
 * Broker endpoint and the thing name should be configured in demos/include/aws_clientcredential.h.
```

File: demos/ble/README.md [hunk 4]
```diff
@@ -11,3 +11,6 @@ This demo creates a special service with two characteristics, one used to count,
...
 This application demonstrates the transport interface implementation with BLE to send MQTT packets to the cloud via a mobile device application. The demo requires a companion mobile application written using the [iOS](https://github.com/aws/amazon-freertos-ble-ios-sdk) or [Android](https://github.com/aws/amazon-freertos-ble-android-sdk) SDK. Sample mobile application can be found in the respective repositories. The demo facilitates applications written using the MQTT library to seamlessly send to and receive data from the cloud, over BLE through a companion device. 
+
+## Shadow BLE Transport Demo
+This application demonstrates shadow updates for a BLE device with the cloud using a companion mobile device application. The demo requires a smartphone application written using the companion SDK for [iOS](https://github.com/aws/amazon-freertos-ble-ios-sdk) or [Android](https://github.com/aws/amazon-freertos-ble-android-sdk). Sample mobile applications can be found in the respective repositories. The demo facilitates applications written using the shadow and MQTT library to seamlessly perform shadow updates with the cloud over BLE through a companion device.
```
Comment (aggarw13) [inline demos/ble/README.md:16]:
```suggestion
This application demonstrates Device Shadow updates for a BLE device with AWS IoT using a companion mobile device application. The demo requires a smartphone application written using the companion SDK for [iOS](https://github.com/aws/amazon-freertos-ble-ios-sdk) or [Android](https://github.com/aws/amazon-freertos-ble-android-sdk). Sample mobile applications can be found in the respective repositories. The demo facilitates applications written using the shadow and MQTT libraries to seamlessly perform shadow updates with the cloud over BLE through a companion device. 
```

File: vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_demos/config_files/shadow_ble_demo_config.h [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://www.FreeRTOS.org
+ * http://aws.amazon.com/freertos
+ *
+ * 1 tab == 4 spaces!
+ */
+
+#ifndef SHADOW_DEMO_CONFIG_H
```
Comment (aggarg) [inline vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_demos/config_files/shadow_ble_demo_config.h:28]: SHADOW_DEMO_CONFIG_H --> SHADOW_BLE_DEMO_CONFIG_H

File: vendors/espressif/boards/esp32/aws_demos/config_files/shadow_ble_demo_config.h [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://www.FreeRTOS.org
+ * http://aws.amazon.com/freertos
+ *
+ * 1 tab == 4 spaces!
+ */
+
+#ifndef SHADOW_DEMO_CONFIG_H
```
Comment (aggarg) [inline vendors/espressif/boards/esp32/aws_demos/config_files/shadow_ble_demo_config.h:28]: Same as above.

File: demos/ble/shadow_ble/shadow_demo_ble_transport.c [hunk 7]
```diff
@@ -0,0 +1,1330 @@
...
+#endif
+
+
+/**
+ * @brief The length of #THING_NAME.
+ */
+#define THING_NAME_LENGTH    ( ( uint16_t ) ( sizeof( THING_NAME ) - 1 ) )
+
+
+static NetworkContext_t xBLETransportCtxt = { 0 };
```
Comment (aggarg) [inline demos/ble/shadow_ble/shadow_demo_ble_transport.c:206]: Missing docs.

File: demos/ble/shadow_ble/shadow_demo_ble_transport.c [hunk 8]
```diff
@@ -0,0 +1,1330 @@
...
+
+static bool channelActive = false;
+
+/**
+ * @brief Static buffer used to queue packets on BLE transport interface.
+ */
+static uint8_t ucTransportBuffer[ democonfigBLE_TRANSPORT_BUFFER_SIZE ] = { 0 };
+
+
+static MQTTContext_t xMQTTContext = { 0 };
```
Comment (aggarg) [inline demos/ble/shadow_ble/shadow_demo_ble_transport.c:218]: Missing docs.

File: demos/ble/shadow_ble/shadow_demo_ble_transport.c [hunk 9]
```diff
@@ -0,0 +1,1330 @@
...
+ * @param[in] pxPacketInfo Packet Info pointer for the incoming packet.
+ * @param[in] pxDeserializedInfo Deserialized information from the incoming packet.
+ */
+static void prvEventCallback( MQTTContext_t * pxMqttContext,
+                              MQTTPacketInfo_t * pxPacketInfo,
+                              MQTTDeserializedInfo_t * pxDeserializedInfo );
+
+
+
+static void bleChannelCallback( IotBleDataTransferChannelEvent_t event,
```
Comment (aggarg) [inline demos/ble/shadow_ble/shadow_demo_ble_transport.c:325]: Add a declaration and docs for all the functions.. Also, if not already done, please run uncrustify.

File: demos/ble/shadow_ble/shadow_demo_ble_transport.c [hunk 10]
```diff
@@ -0,0 +1,1330 @@
...
+                                IotBleDataTransferChannel_t * pChannel,
+                                void * context )
+{
+    MQTTStatus_t acceptCode;
```
Comment (aggarg) [inline demos/ble/shadow_ble/shadow_demo_ble_transport.c:329]: Does not seem to be used. Do we not get unused variable warning?

File: demos/ble/shadow_ble/shadow_demo_ble_transport.c [hunk 11]
```diff
@@ -0,0 +1,1330 @@
...
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t prvBLETransportInterfaceDisconnect( NetworkContext_t * pTransportCtxt )
+{
+    int32_t status = EXIT_SUCCESS;
+
+    /* Open is a handshake proceture, so we need to wait until it is ready to use. */
```
Comment (aggarg) [inline demos/ble/shadow_ble/shadow_demo_ble_transport.c:413]: Does the comment belong here?

File: demos/ble/shadow_ble/shadow_demo_ble_transport.c [hunk 12]
```diff
@@ -0,0 +1,1330 @@
...
+            LogWarn( ( "PINGRESP should not be handled by the application "
+                       "callback when using MQTT_ProcessLoop.\n\n" ) );
+            break;
+
+        case MQTT_PACKET_TYPE_PUBACK:
+            LogInfo( ( "PUBACK received for packet id %u.\n\n",
+                       usPacketIdentifier ) );
+            assert( globalPublishPacketIdentifier == usPacketIdentifier );
+            pubAckReceived = true;
+            /* TODO assert the packet identifier */
```
Comment (aggarg) [inline demos/ble/shadow_ble/shadow_demo_ble_transport.c:455]: TODO can be removed?

File: demos/ble/shadow_ble/shadow_demo_ble_transport.c [hunk 13]
```diff
@@ -0,0 +1,1330 @@
...
+        returnStatus = EXIT_FAILURE;
+    }
+    else
+    {
+        LogInfo( ( "UNSUBSCRIBE sent topic %.*s to broker.\n\n",
+                   usTopicFilterLength,
+                   pcTopicFilter ) );
+
+        /* Process incoming packet from the broker. Acknowledgment for subscription
+         * ( SUBACK ) will be received here. However after sending the subscribe, the
```
Comment (aggarg) [inline demos/ble/shadow_ble/shadow_demo_ble_transport.c:986]:
```suggestion
        /* Process incoming packet from the broker. Acknowledgment for unsubscribe operation
         * ( UNSUBACK ) will be received here. However after sending the subscribe, the
```

File: demos/ble/shadow_ble/shadow_demo_ble_transport.c [hunk 14]
```diff
@@ -0,0 +1,1330 @@
...
+ * - SHADOW_TOPIC_STRING_UPDATE_ACCEPTED for "$aws/things/thingName/shadow/update/accepted"
+ * - SHADOW_TOPIC_STRING_UPDATE_REJECTED for "$aws/things/thingName/shadow/update/rejected"
+ *
+ * It also uses these macros for topics to publish to:
+ * - SHADOW_TOPIC_STIRNG_DELETE for "$aws/things/thingName/shadow/delete"
+ * - SHADOW_TOPIC_STRING_UPDATE for "$aws/things/thingName/shadow/update"
+ *
+ * The helper functions this demo uses for MQTT operations have internal
+ * loops to process incoming messages. Those are not the focus of this demo
+ * and therefor, are placed in a separate file shadow_demo_helpers.c.
```
Comment (aggarg) [inline demos/ble/shadow_ble/shadow_demo_ble_transport.c:1122]: Is the comment about functions in a separate file accurate?

File: vendors/espressif/boards/esp32/aws_demos/config_files/shadow_ble_demo_config.h (hunk 6 above)
Comment (ravibhagavandas) [reply vendors/espressif/boards/esp32/aws_demos/config_files/shadow_ble_demo_config.h:28]: Fixed

**Review by aggarg — DISMISSED** [review]

File: vendors/espressif/boards/esp32/aws_demos/config_files/shadow_ble_demo_config.h [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+ * #define democonfigMQTT_BROKER_ENDPOINT    "...insert here..."
+ */
+
+/**
+ * @brief The port to use for the demo.
+ *
+ * In general, port 8883 is for secured MQTT connections.
+ *
+ * @note Port 443 requires use of the ALPN TLS extension with the ALPN protocol
+ * name. When using port 8883, ALPN is not required.
```
Comment (aggarw13) [inline vendors/espressif/boards/esp32/aws_demos/config_files/shadow_ble_demo_config.h:88]: Is this information relevant to the BLE demos?

Comment (ravibhagavandas) [reply vendors/espressif/boards/esp32/aws_demos/config_files/shadow_ble_demo_config.h]: Removed unused comments

File: projects/cypress/CY8CKIT_064S0S2_4343W/mtb/aws_demos/afr.mk [hunk 16]
```diff
@@ -363,7 +363,6 @@ SOURCES+=\
 	$(wildcard $(CY_AFR_ROOT)/libraries/abstractions/ble_hal/*c)\
 	$(wildcard $(CY_AFR_ROOT)/demos/ble/gatt_server/*.c)\
 	$(wildcard $(CY_AFR_ROOT)/demos/ble/mqtt_ble/*.c)\
-	$(wildcard $(CY_AFR_ROOT)/demos/ble/shadow_ble/*.c)\
```
Comment (aggarw13) [inline projects/cypress/CY8CKIT_064S0S2_4343W/mtb/aws_demos/afr.mk]: Why was this removed?

Comment (ravibhagavandas) [reply projects/cypress/CY8CKIT_064S0S2_4343W/mtb/aws_demos/afr.mk]: Added after rebase.

**Review by aggarw13 — APPROVED** [review]

**Review by aggarg — APPROVED** [review]
